get_complete_position supports imports and panic handling

nhooyr opened the pull request:
This closes #46 and #45

nhooyr committed:
get_complete_position supports imports

I also added better handling of gocode panics
This closes #46 and #45

nhooyr commented:
Actually don't merge, I think I can make it a bit more clean.

zchee commented:
@nhooyr OK :)

nhooyr commented:
@zchee you might know how to do this.

If we get the string `"github.com/` with `re.search` I want to match `"github.com` but make `re.start()` act like the pattern started at `g`. So if I have the following:

`"github.com|`

I want the regex to match the entire string with the `"` but make re.start() return the position at g. 

`"|github.com` not `|"github.com`.

Obviously this shouldn't happen for patterns like `fmt.Println`. So I have two regexes, one that matches the import statement, other that matches normal statements. Instead I can do it in a single statement if python regexes supports this.

nhooyr commented:
actually never mind, its fine.

zchee commented on rplugin/python3/deoplete/sources/go.py:
```diff
@@ -34,8 +34,12 @@ def __init__(self, vim):
     def get_complete_position(self, context):
-        m = re.search(r'\w*$', context['input'])
-        return m.start() if m else -1
+        m = re.search(r'"[^"]*$', context['input'])
```
Maybe, this regex is better than it. (slightly tested)

``` python
m = re.search(r'(\w*$|(?<=")([\.\/\w]*$))', context['input'])
```

Not required `if...else` if use this regex.
Could you test it?

nhooyr replied:
yup, thats what I was asking about, in my big post. how to do that in regex.

nhooyr commented:
@zchee That regex you commented is exactly what I was asking about. I just didn't know the syntax in python. I cleaned it up a bit and updated the PR.

zchee commented:
@nhooyr LGTM. Thanks!!
